sqlcon/connector_test.go

```diff
@@ -330,6 +342,38 @@ func bootstrapPostgres() {
 	postgresURL = u
 }
 
+func bootstrapCockroach() {
+	if uu := os.Getenv("TEST_DATABASE_COCKROACHDB"); uu != "" {
+		log.Println("Found cockroachdb test database config, skipping dockertest...")
+		_, err := sqlx.Open("postgres", uu)
+		if err != nil {
+			log.Fatalf("Could not connect to bootstrapped database: %s", err)
+		}
+		u, _ := url.Parse(uu)
+		cockroachURL = u
+		return
+	}
+
+	pool, err := dockertest.NewPool("")
+	if err != nil {
+		log.Fatalf("Could not Connect to docker: %s", err)
+	}
+
+	resource, err := pool.RunWithOptions(&dockertest.RunOptions{
+		Repository: "cockroachdb/cockroach",
+		Tag:        "v2.1.6",
+		Cmd:        []string{"start --insecure"},
+	})
+	if err != nil {
+		log.Fatalf("Could not start resource: %s", err)
+	}
+
+	urls := bootstrap("postgres://root@localhost:%s/defaultdb?sslmode=disable", "26257/tcp", "postgres", pool, resource)
```

Comment:
It can't be cockroach, because we need the "real" dsn (from GO driver perspective) inside the bootstrap function, for a successful `sql.Open` to happen.

I just applied the same logic it's been applied with mysql (send the DSN without `mysql://`, then add it after the bootstrap function).

Unless I'm missing something here the only thing we can do it's just change the place of the replace to inside the boostrap function with a switch/case (for mysql and cockroach) instead of doing it individually inside  `bootstrapMySQL()` and `bootstrapCockroach()`.

See here:

https://github.com/ory/x/blob/master/sqlcon/connector_test.go#L299-L302

And here:

https://github.com/ory/x/blob/master/sqlcon/connector_test.go#L337